csharp/ql/src/Likely Bugs/StaticFieldWrittenByInstance.ql

```diff
@@ -14,12 +14,12 @@
 
 import csharp
 
-from FieldWrite fw, Field f, Callable m
+from FieldWrite fw, Field f, Callable c
 where
   fw.getTarget() = f and
   f.isStatic() and
-  m = fw.getEnclosingCallable() and
-  not m.(Member).isStatic() and
-  f.getDeclaringType() = m.getDeclaringType() and
-  m.fromSource()
-select fw.(VariableAccess), "Write to static field from instance method or constructor."
+  c = fw.getEnclosingCallable() and
+  not exists(Member m | m = c or m = c.(Accessor).getDeclaration() | m.isStatic()) and
```

Comment:
Question: could this be implemented like this:
```suggestion
  not (c.(Member).isStatic() or c.(Accessor).getDeclaration().isStatic()) and
```
and would that (potentially) be better for performance?

Reply:
Or even shorter
```ql
not [c.(Member), c.(Accessor).getDeclaration()].isStatic()
```

Reply:
Yes, it can be implemented like this, but I don't think that it will have any effect on performance.
I prefer the original solution as it re-uses the `isStatic` logic across the relevant members.